internal/scorecard/alpha/formatting.go

```diff
@@ -19,45 +19,38 @@ import (
 	"fmt"
 
 	"github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
-// getTestResults fetches the test pod logs and converts it into
+// getTestResult fetches the test pod log and converts it into
 // ScorecardOutput format
-func getTestResults(client kubernetes.Interface, tests []Test) (output v1alpha2.ScorecardOutput) {
-	output.Results = make([]v1alpha2.ScorecardTestResult, 0)
+func getTestResult(client kubernetes.Interface, p *v1.Pod, test Test) (output v1alpha2.ScorecardTestResult) {
 
-	for _, test := range tests {
-		p := test.TestPod
-		logBytes, err := getPodLog(client, p)
-		if err != nil {
-			r := v1alpha2.ScorecardTestResult{}
-			r.Name = test.Name
-			r.Description = test.Description
-			r.Errors = []string{fmt.Sprintf("Error getting pod log %s", err.Error())}
-			output.Results = append(output.Results, r)
-		} else {
-			// marshal pod log into ScorecardTestResult
-			var sc v1alpha2.ScorecardTestResult
-			err := json.Unmarshal(logBytes, &sc)
-			if err != nil {
-				r := v1alpha2.ScorecardTestResult{}
-				r.Name = test.Name
-				r.Description = test.Description
-				r.Errors = []string{fmt.Sprintf("Error unmarshalling test result %s", err.Error())}
-				output.Results = append(output.Results, r)
-			} else {
-				output.Results = append(output.Results, sc)
-			}
-		}
+	logBytes, err := getPodLog(client, p)
+	if err != nil {
+		r := v1alpha2.ScorecardTestResult{}
+		r.Name = test.Name
+		r.Description = test.Description
+		r.Errors = []string{fmt.Sprintf("Error getting pod log %s", err.Error())}
+		return r
+	}
+	// marshal pod log into ScorecardTestResult
+	err = json.Unmarshal(logBytes, &output)
+	if err != nil {
+		r := v1alpha2.ScorecardTestResult{}
+		r.Name = test.Name
+		r.Description = test.Description
+		r.Errors = []string{fmt.Sprintf("Error unmarshalling test result %s", err.Error())}
+		return r
 	}
 	return output
 }
 
 // ListTests lists the scorecard tests as configured that would be
 // run based on user selection
 func (o Scorecard) ListTests() (output v1alpha2.ScorecardOutput, err error) {
-	tests := selectTests(o.Selector, o.Config.Tests)
+	tests := o.selectTests()
 	if len(tests) == 0 {
 		fmt.Println("no tests selected")
```

Comment:
If something is parsing the output as JSON, this will break it. We need to handle this type of output in the output formatter.

Reply:
done